dateutil/parser/isoparser.py

```diff
@@ -206,7 +206,7 @@ def _parse_isodate_common(self, dt_str):
         pos += 2
 
         if pos >= len_str:
-            return components, pos
+            raise ValueError('YYYYMM or YYYY-MM is an invalid ISO 8601 format')
```

Comment:
`YYYY-MM` is very valid. It is *only* `YYYYMM` that is not valid (because it creates an ambiguity between ISO-8601 dates and non-ISO8601 `YYMMDD` triplets.